### ISSUE TITLE
Some problems with converting Tensorflow framework to PyTorch framework

### ISSUE THREAD

**Issue by chenxl124578** [itailang/SampleNet#16]
Question: Is there any difference of the network or loss between the registration task and the classification task.

Description: I want to change the classification task of SampleNet's code to PyTorch framework(like the example code snippet in README.md). PointNet classifier is working normally(get a 90 eval acc pretrained model). But I got a terrible training result in SampleNet. Like train acc (in last epoch=500) =23% and eval acc = 61%. I check my code and didn't found error in  calculate acc part. I don't know Is there any difference of **the network or loss** between the registration task and the classification task , because I haven't used Tensorflow in the past.

**Comment by itailang** [comment]
Dear @chenxl124578,

Thank you for your interest in our work. The hyperparameters are different for the classification and registration tasks. Please see Table 6 in the supplementary material of the paper.

**Comment by chenxl124578** [comment]
Thanks a lot! I have solved the problem by setting alpha=0.01 in Loss, and get a good result. But there is another question that why alpha=30 work in Tensorflow and alpha=0.01 work in PyTorch. 
That's really an interesting question may be i need to do some experiments to find the reason.

**Comment by itailang** [comment]
Dear @chenxl124578,

I am glad to know that you solved the problem!

It is interesting indeed. I would check if there is a difference in the architecture or the training procedure of the classification network in TensorFlow and PyTroch.

Anyway, if you find the reason for the change, you are welcome to share your findings here. Thank you!

**Comment by chenxl124578** [comment]
Hi, i have done some experiment with your code, but now get some question about the Tensorflow.

In S-NET paper's Figure 11, you describe that "Without the regularization, the accuracy using the generated points is slightly increased". And the reason is "since the regularization sacrifices some of the classification loss to minimize the sampling loss."

But in SampleNet, when i set alpha=0.01(near 0), i got 70acc in eval(not inference). And i also got 83acc with alpha=30. This is a little different from the conclusion that acc(generated point) will increase without regularization loss.

So I wonder whether the eval part of the train_samplenet.py is the same forward process as the train part. (I usually use PyTorch and am not familiar with tensorflow1. 😢)

Looking forward to your reply!

**Comment by itailang** [comment]
Dear @chenxl124578,

In SanpleNet, the generated points do not participate in the loss, but rather the approximated sampled points. This is different than in S-NET, where the generated point are routed to the loss. Thus, with weaker regularization in SampleNet, the generated point are less similar to the sampled points, which are optimized to the task, and the performance with the generated points degrades.
